docs/apidocs/transpiler_builtin_plugins.rst

```diff
@@ -0,0 +1,10 @@
+.. _qiskit-transpiler-builtin-plugins:
+
+====================================
+Builtin Transpiler Synthesis Plugins
+====================================
+
+.. toctree::
+   :maxdepth: 2
+
+   synthesis_aqc
```

Comment:
Yeah, I didn't put it in here because it didn't have any user-facing documentation to link to.  If I make a stub file, it's basically just going to be the text "The default unitary synthesis plugin".